tests/unit/system/classes/MediaLibraryTest.php

```diff
@@ -62,4 +68,60 @@ public function testValidPathsOnValidatePath($path)
         $result = MediaLibrary::validatePath($path);
         $this->assertInternalType('string', $result);
     }
+
+    public function testListFolderContents()
+    {
+        $this->copyMedia();
+
+        $contents = MediaLibrary::instance()->listFolderContents();
+        $this->assertNotEmpty($contents, 'Media library item is not discovered');
+
+        $item = reset($contents);
+        $this->assertAttributeEquals('file', 'type', $item, 'Media library item does not have the right type');
+        $this->assertAttributeEquals('/text.txt', 'path', $item, 'Media library item does not have the right path');
+        $this->assertAttributeNotEmpty('lastModified', $item, 'Media library item last modified is empty');
+        $this->assertAttributeNotEmpty('size', $item, 'Media library item size is empty');
+
+        $this->removeMedia();
+    }
+
+    protected function setUpStorage()
+    {
+        $this->app->useStoragePath(base_path('storage/temp'));
+
+        config(['filesystems.disks.test_local' => [
+            'driver' => 'local',
+            'root'   => storage_path('app'),
+        ]]);
+
+        config(['cms.storage.media' => [
+            'disk'   => 'test_local',
+            'folder' => 'media',
+            'path'   => '/storage/app/media',
+        ]]);
+    }
+
+    protected function copyMedia()
+    {
+        $mediaPath = storage_path('app/media');
+
+        if (!is_dir($mediaPath)) {
+            mkdir($mediaPath, 0777, true);
+        }
+
+        foreach (glob(base_path('tests/fixtures/media/*')) as $file) {
+            $path = pathinfo($file);
+            copy($file, $mediaPath . DIRECTORY_SEPARATOR . $path['basename']);
+        }
+    }
+
+    protected function removeMedia()
+    {
+        foreach (glob(storage_path('app/media/*')) as $file) {
+            unlink($file);
+        }
+
+        rmdir(storage_path('app/media'));
+        rmdir(storage_path('app'));
```

Comment:
@DieterHolvoet just to clarify, this media removal all happens within your temp storage folder, yes? It doesn't have any effect on the actual media folder?

Reply:
Yes, the storage path is changed by calling
```php
$this->app->useStoragePath(base_path('storage/temp'));
```
so there's no effect on the usual media folder.

Reply:
I added an extra test that makes sure files are only deleted when the storage path is changed.